src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AuditCategory.java

```diff
@@ -22,19 +23,18 @@
     COMPLIANCE_INTERNAL_CONFIG_READ,
     COMPLIANCE_INTERNAL_CONFIG_WRITE;
 
-    public static EnumSet<AuditCategory> parse(final Collection<String> categories) {
-        EnumSet<AuditCategory> set = EnumSet.noneOf(AuditCategory.class);
+    public static Set<AuditCategory> parse(final Collection<String> categories) {
         if (categories.isEmpty())
-            return set;
+            return Collections.emptySet();
 
         return categories
                 .stream()
                 .map(String::toUpperCase)
                 .map(AuditCategory::valueOf)
-                .collect(Collectors.toCollection(() -> set));
+                .collect(ImmutableSet.toImmutableSet());
```

Comment:
Which order is important here and why? Should not natural order be sufficient?

Reply:
If the entries are passed as array, then the enum set reorders based on the order of elements defined within EnumSet. 
For eg: `["AUTHENTICATED", "GRANTED_PRIVILEGES"]` gets reordered as `["GRANTED_PRIVILEGES", "AUTHENTICATED"]` as `GRANTED_PRIVILEGES comes` before `AUTHENTICATED `in EnumSet ordering. 

This breaks if the same put request is called twice because to compare two JSON arrays are equal, the order must be same too. Also in the UI, the elements gets reordered on refetching. To avoid this, using ImmutableSet so that ordering remains the same as the way user enters them.